fast_matched_filter/src/matched_filter.cu

```diff
@@ -133,7 +133,7 @@ void matched_filter(float *templates, float *sum_square_templates,
 
     // find the number of available GPUs
     cudaGetDeviceCount(&nGPUs);
-    omp_set_num_threads(nGPUs);
+    omp_set_num_threads((int)fminf((float)nGPUs, (float)n_templates));
```

Comment:
Any reason not to use integer min function? Not that this is time critical, but it looks like CUDA supports `min()` and `max()` - I haven't tested, but just googled and [found this](https://devtalk.nvidia.com/default/topic/486202/max-a-b-a-62-b-a-b-in-cuda-how-to-use-above-c-function-in-cuda/) with the comment:
> CUDA already supports min() and max() functions. These are overloaded functions with support for a large number of types, so there should be no need to define macros for this purpose. Note that functions are safer than macros for this purpose since the arguments are not evaluated twice. In addition to overloaded min() and max() there are also the C99-compatible math functions fmin(), fmax(), fminf(), fmaxf() for floating-point data

Reply:
How about:
```C
omp_set_num_threads(min(nGPUS, (int)n_templates));
```
Then you only have one cast (from `size_t` to `int`)?  Otherwise I tested this with our GPU machine and instead of using all 4 GPUs it used 1, so happy days!

Reply:
Sorry, I completely missed getting alerts for your comments. Yes, I agree, this looks like the route to go!